packages/components/src/VirtualizedList/RowSelection/RowSelection.component.js

```diff
@@ -7,18 +7,21 @@ import theme from './RowSelection.scss';
  * Higher order row-renderer that wrap the provided row renderer.
  * It manages row selection classname and inject it to the row renderer props.
  */
-function getRowSelectionRenderer(WrappedComponent, { isSelected, getRowData }) {
+function getRowSelectionRenderer(WrappedComponent,
+	{ isSelected, isActive, getRowData, onRowClick }) {
 	function RowSelection(props) {
 		const rowData = getRowData(props);
 
 		const enhancedClassNames = classNames(
 			props.className,
 			theme['row-selection'],
-			{ [theme.selected]: isSelected(rowData) },
-			{ selected: isSelected(rowData) },
+			{ [theme.active]: isActive && isActive(rowData) },
+			{ active: isActive && isActive(rowData) },
+			{ [theme.selected]: isSelected && isSelected(rowData) },
+			{ selected: isSelected && isSelected(rowData) },
 		);
 		return (
-			<WrappedComponent {...props} className={enhancedClassNames} />
+			<WrappedComponent {...props} onRowClick={onRowClick} className={enhancedClassNames} />
```

Comment:
This is not this HOC responsibility to pass the onRowClick. It is only supposed to handle active/selection style.
I assume you've done that for Large display. Indeed onRowClick is not something passed to the RowRenderer, but from it you have access to its parent.
Remove that and see my comment on RowLarge.component.js